src/TorchSharp/JIT/ScriptModule.cs

```diff
@@ -697,22 +793,62 @@ public static ScriptModule<TResult> load<TResult>(string filename)
+            /// Load a ScriptModule or ScriptFunction previously saved with torch.jit.save
+            /// </summary>
+            /// <typeparam name="T1">The first argument type.</typeparam>
+            /// <typeparam name="T2">The second argument type.</typeparam>
+            /// <typeparam name="TResult">The return type of the module.</typeparam>
+            /// <param name="filename">The file name of the module.</param>
+            /// <param name="map_location">The device type where the script module should be loaded.</param>
+            /// <returns>A ScriptModule instance, whether the script originated as a module or function.</returns>
+            /// <remarks>
+            /// All previously saved modules, no matter their device, are first loaded onto CPU, and then are moved to the devices they were saved from.If this fails (e.g.because the run time system doesn’t have certain devices), an exception is raised.
+            /// </remarks>
+            public static ScriptModule<T1, T2, TResult> load<T1, T2, TResult>(string filename, Device map_location)
+            {
+                return new ScriptModule<T1, T2, TResult>(_load(filename, map_location.type, map_location.index));
+            }
+
+            /// <summary>
+            /// Load a ScriptModule or ScriptFunction previously saved with torch.jit.save
+            /// </summary>
+            /// <typeparam name="T1">The first argument type.</typeparam>
+            /// <typeparam name="T2">The second argument type.</typeparam>
+            /// <typeparam name="TResult">The return type of the module.</typeparam>
+            /// <param name="filename">The file name of the module.</param>
+            /// <param name="map_location">The device type where the script module should be loaded.</param>
+            /// <returns>A ScriptModule instance, whether the script originated as a module or function.</returns>
+            /// <remarks>
+            /// All previously saved modules, no matter their device, are first loaded onto CPU, and then are moved to the devices they were saved from.If this fails (e.g.because the run time system doesn’t have certain devices), an exception is raised.
+            /// </remarks>
+            public static ScriptModule<T1, T2, TResult> load<T1, T2, TResult>(string filename, string map_location)
+            {
+                return load<T1, T2, TResult>(filename, new Device(map_location));
+            }
+
+            private static IntPtr _load(string filename, DeviceType device_type, long device_index)
             {
                 if (!System.IO.File.Exists(filename))
                     throw new System.IO.FileNotFoundException(filename);
 
-                var result = THSJIT_load(filename);
+                if (device_type != DeviceType.CUDA) device_index = -1;
+
+                if (device_type == DeviceType.CUDA && !torch.cuda.is_available()) throw new InvalidOperationException("CUDA is not available.");
```

Comment:
nit: this may be written as the following?
```C
 if (device_type != DeviceType.CUDA) device_index = -1;
else if (!torch.cuda.is_available()) throw new InvalidOperationException("CUDA is not available.");
```